lang/src/main/kotlin/org/partiql/lang/eval/ExprValueFactory.kt

```diff
@@ -184,11 +167,11 @@ private class ExprValueFactoryImpl(override val ion: IonSystem) : ExprValueFacto
         private val MIN_LONG_VALUE = BigInteger.valueOf(Long.MIN_VALUE)
     }
 
-    override val missingValue = MissingExprValue(ion.newNull())
-    override val nullValue = NullExprValue(ion.newNull())
+    override val missingValue = MissingExprValue(ion)
+    override val nullValue = NullExprValue(ion)
 
-    private val trueValue = TrueBoolExprValue(ion.newBool(true))
-    private val falseValue = FalseBoolExprValue(ion.newBool(false))
+    private val trueValue = TrueBoolExprValue(ion)
+    private val falseValue = FalseBoolExprValue(ion)
```

Comment:
I guess this a question for a different place, but why do we have two different True/False bool values? Is this because they're "atomic" and every "false" is the same "false"? Seems weird this way rather than a traditional constants in the top-level (BoolExprValue) class.

```kotlin
// private constructor, control instantiation
class BoolExprValue(private val value: Boolean) private constructor()

     companion object {
   
         // you can still instantiate from a primitive, but the authors have control
          fun of(value: Boolean) = when (value) {
              true -> BoolExprValue.TRUE
              false -> BoolExprValue.FALSE
          }
    }

// ... naming(?)
BoolExprValue.TRUE     // BoolExprValue(true)
BoolExprValue.FALSE    // BoolExprValue(false)
```

Reply:
Why is this different? The public API for constructing boolean values is `ExprValueFactory.newBool(value: Boolean)`